js/classes/entities/midground/transport_portal.js

```diff
@@ -0,0 +1,133 @@
+class TransportPortal {
+    /**
+     * @param {Vector} blockPos The position of the block the portal is on
+     * @param {Number} type The type of portal (e.g. TransportPortal.PURPLE or TransportPortal.YELLOW)
+     * @author Nathan Hinthorne
+     */
+    constructor(blockPos, type) {
+        this.pos = Vector.blockToWorldSpace(blockPos);
+        this.type = type;
+
+        /** The number of enemies to release at once. */
+        this.spawnGroupSize = TransportPortal.SPAWN_GROUP_SIZE;
+
+        this.animations = [];
+        this.loadAnimations();
+
+        /** Whether the portal is active, in the middle of activating, or inactive */
+        this.action = "inactive";
+
+        switch (this.type) {
+            case TransportPortal.PURPLE:
+                this.particleType = ParticleEffect.AURA_PURPLE;
+                this.borderColor = COLORS.PURPLE;
+                break;
+            case TransportPortal.YELLOW:
+                this.particleType = ParticleEffect.AURA_YELLOW;
+                this.borderColor = COLORS.YELLOW;
+                break;
+        }
+
+        this.activationTimer = TransportPortal.ACTIVATION_DELAY;
+        this.particleTimer = 0;
+        this.center = Vector.add(this.pos, new Vector(TransportPortal.SCALED_SIZE.x / 2, TransportPortal.SCALED_SIZE.y / 2));
+    };
+
+    activate() {
+        this.action = "activating";
+        ASSET_MGR.playSFX(SFX.PORTAL_ACTIVATE.path, SFX.PORTAL_ACTIVATE.volume);
```

Comment:
I noticed that when you go through a portal multiple times quickly the audio repeats each time. Could we have the playSFX stop the currently playing sfx to play the new one if it's a portal or an sfx that should only play once at a time?

Reply:
I'll see what I can do. It might be a complicated fix.